stable_baselines3/common/policies.py

```diff
@@ -114,16 +114,18 @@ def make_features_extractor(self) -> BaseFeaturesExtractor:
         """Helper method to create a features extractor."""
         return self.features_extractor_class(self.observation_space, **self.features_extractor_kwargs)
 
-    def extract_features(self, obs: th.Tensor) -> th.Tensor:
+    def extract_features(self, obs: th.Tensor, features_extractor: Optional[BaseFeaturesExtractor] = None) -> th.Tensor:
         """
         Preprocess the observation if needed and extract features.
 
-        :param obs:
-        :return:
+         :param obs: The observation
+         :param features_extractor: The features extractor to use. If it is set to None, the features extractor of the policy is used.
+         :return: The features
         """
-        assert self.features_extractor is not None, "No features extractor was set"
+        features_extractor = features_extractor or self.features_extractor
+        assert features_extractor is not None, "No features extractor was set"
         preprocessed_obs = preprocess_obs(obs, self.observation_space, normalize_images=self.normalize_images)
-        return self.features_extractor(preprocessed_obs)
+        return features_extractor(preprocessed_obs)
```

Comment:
I'm not very satisfied with this method, when you see it it's not easy to understand why in some cases you need to use the feature extractor of the instance, and in others not. I think that in the future it should be simplified.

Reply:
> I think that in the future it should be simplified.

how would you simplify it? 
always call it with a `features_extractor` parameter?

Reply:
> > I think that in the future it should be simplified.
> 
> how would you simplify it? always call it with a `features_extractor` parameter?

Yeah I personally think it could be a bit clearer to always pass the features extractor to that method, both in case it corresponds to `self.features_extractor` and in case it's a `pi` or `vf` features extractor.

I could do it, in case you opt for that

Reply:
@qgallouedec I added deprecation warning so we can make it the default in v1.8.0.